### PR TITLE
Moved grouptest over to pure unit tests

### PR DESCRIPTION
**Pull request by rullzer**
Now that we have a non static SubAdmin class we can start moving code that relied on that to pure unit tests! :smile: 

100%  test coverage
Let see if integration tests are still happy ;)

CC: @PVince81 @MorrisJobke @LukasReschke

### CODE REVIEW

File: apps/provisioning_api/lib/groups.php [hunk 1]
```diff
@@ -54,9 +59,16 @@ public function __construct(\OCP\IGroupManager $groupManager,
 	 * @return OC_OCS_Result
 	 */
 	public function getGroups($parameters) {
-		$search = !empty($_GET['search']) ? $_GET['search'] : '';
-		$limit = !empty($_GET['limit']) ? $_GET['limit'] : null;
-		$offset = !empty($_GET['offset']) ? $_GET['offset'] : null;
+		$search = $this->request->getParam('search');
```
Comment (MorrisJobke) [inline apps/provisioning_api/lib/groups.php]: ProTip: You can also specify a default value ;)

File: apps/provisioning_api/lib/groups.php [hunk 2]
```diff
@@ -161,14 +173,8 @@ public function getSubAdminsOfGroup($parameters) {
 		foreach ($subadmins as $user) {
 			$uids[] = $user->getUID();
 		}
-		$subadmins = $uids;
 
-		// Go
-		if(!$subadmins) {
```
Comment (DeepDiver1975) [inline apps/provisioning_api/lib/groups.php]: this logic is missing in the new code - reasons? This is a change in behavior

Comment (rullzer) [inline apps/provisioning_api/lib/groups.php]: Ah yes, I figured this was an old leftover. It makes no sense to throw an error (especially "Unkown error occured") if there are no subadmins. Just an empty list would be the expected result then. No subadmins are not an error case.

Comment (rullzer) [inline apps/provisioning_api/lib/groups.php]: But you are right maybe i should have made that a different PR

File: apps/provisioning_api/lib/groups.php (hunk 1 above)
Comment (rullzer) [inline apps/provisioning_api/lib/groups.php]: Ah yes fixed

File: apps/provisioning_api/lib/groups.php (hunk 2 above)
Comment (rullzer) [inline apps/provisioning_api/lib/groups.php]: Ok fixed. Not messing with the behaviour now.

**Comment by MorrisJobke** [comment]
I rebased in current master -> lets see if this still passes CI :)

**Comment by MorrisJobke** [comment]
beside that it looks good :+1:

**Comment by rullzer** [comment]
Rebases. Just one more review ;)

CC: @PVince81 @LukasReschke @DeepDiver1975 @tomneedham

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**